dali/test/python/test_operator_audio_resample.py

```diff
@@ -34,64 +34,73 @@
+    if device == 'gpu':
+      audio0 = audio0.gpu()
     audio2 = fn.experimental.audio_resample(audio0, in_rate=sr0, out_rate=out_sr)
     audio3 = fn.experimental.audio_resample(audio0, scale=out_sr/sr0)
     audio4 = fn.experimental.audio_resample(audio0, out_length=fn.shapes(audio1)[0])
     return audio1, audio2, audio3, audio4
 
-def test_standalone_vs_fused():
-    pipe = audio_decoder_pipe(batch_size=2, num_threads=1, device_id=0)
+def _test_standalone_vs_fused(device):
+    pipe = audio_decoder_pipe(device=device, batch_size=2, num_threads=1, device_id=0)
     pipe.build()
+    is_gpu = device == 'gpu'
     for _ in range(2):
         outs = pipe.run()
         # two sampling rates - should be bit-exact
-        check_batch(outs[0], outs[1], eps=0, max_allowed_error=0)
+        check_batch(outs[0], outs[1], eps=1e-6 if is_gpu else 0,
+                    max_allowed_error=1e-4 if is_gpu else 0)
         # numerical round-off error in rate
         check_batch(outs[0], outs[2], eps=1e-6, max_allowed_error=1e-4)
         # here, the sampling rate is slightly different, so we can tolerate larger errors
         check_batch(outs[0], outs[3], eps=1e-4, max_allowed_error=1)
 
-def _test_type_conversion(src_type, in_values, dst_type, out_values, eps):
+def test_standalone_vs_fused():
+    for device in ('gpu', 'cpu'):
+      yield _test_standalone_vs_fused, device
+
+def _test_type_conversion(device, src_type, in_values, dst_type, out_values, eps):
   src_nptype = dali_type_to_np(src_type)
   dst_nptype = dali_type_to_np(dst_type)
   assert len(out_values) == len(in_values)
   in_data = [np.full((100 + 10 * i,), x, src_nptype) for i, x in enumerate(in_values)]
   @pipeline_def(batch_size=len(in_values))
-  def test_pipe():
-    input = fn.external_source(in_data, batch = False, cycle = 'quiet')
+  def test_pipe(device):
+    input = fn.external_source(in_data, batch=False, cycle='quiet', device=device)
     return fn.experimental.audio_resample(input, dtype=dst_type, scale=1, quality=0)
-  pipe = test_pipe(device_id=0, num_threads=4)
+  pipe = test_pipe(device, device_id=0, num_threads=4)
   pipe.build()
+  is_gpu = device == 'gpu'
   for _ in range(2):
     out, = pipe.run()
     assert len(out) == len(out_values)
     assert out.dtype == dst_type
     for i in range(len(out_values)):
       ref = np.full_like(in_data[i], out_values[i], dst_nptype)
-      if not np.allclose(out.at(i), ref, 1e-6, eps):
-        print("Actual: ", out.at(i))
-        print(out.at(i).dtype, out.at(i).shape)
+      out_arr = as_array(out[i])
+      if not np.allclose(out_arr, ref, 1e-6, eps):
+        print("Actual: ", out_arr)
+        print(out_arr.dtype, out_arr.shape)
         print("Reference: ", ref)
         print(ref.dtype, ref.shape)
-        print("Diff: ", out.at(i).astype(np.float) - ref)
-        assert np.allclose(out.at(i), ref, 1e-6, eps)
-
+        print("Diff: ", out_arr.astype(np.float) - ref)
+        assert False
```

Comment:
Nitpick: the original code deliberately repeated the check here, so that the error in nosetests would appear as more than `False` in non-verbose runs.